### ISSUE TITLE
invalid-return-type false positive when using typing.Self with enum

### ISSUE THREAD

**Issue by correctmost** [astral-sh/ty#2915]
### Summary

mypy recently added support for using `Self` with implicitly-final enums: https://github.com/python/mypy/issues/18345

Will ty support this use as well? 

Support in other type checkers:
- Pyright does not seem like it will allow it: https://github.com/microsoft/pyright/issues/7712
- Pyrefly doesn't currently allow it, but I didn't see a specific issue in their bug tracker
- zuban allows it

Example code from the linked mypy issue:
```python
from enum import Enum
from typing import Self

class Foo(Enum):
    BAR = "bar"

    @classmethod
    def get_bar(cls) -> Self:
        return cls.BAR  # error[invalid-return-type] Return type does not match returned value: expected `Self@get_bar`, found `Foo`
```
-- https://play.ty.dev/197e26b2-4e65-41ef-aadb-5ad919b3a117

---

It seems like there has been disagreement in the typing community about this topic, so I wasn't sure if `Self` should continue to be avoided in these enum contexts.

### Version

ty 0.0.19

**Comment by sharkdp** [comment]
Thank you for reporting this.

I think this might be defensible for singleton enums (with a single member), because in those cases `Literal[Foo.BAR]` is the same type as `Foo`. But for enums with multiple members, this is problematic because `Self` may specialize to an enum literal type:
```py
from enum import Enum
from typing import Self

class Answer(Enum):
    YES = "yes"
    NO = "no"

    def return_yes(self) -> Self:
        return Answer.YES  # ty: ignore

reveal_type(Answer.NO.return_yes())  # has type `Literal[Answer.NO]`, but YES at runtime
```
So in case of multi-member enums, we would have to at least perform literal promotion `Literal[Answer.NO]` -> `Answer`?

**Comment by carljm** [comment]
It seems like the use case here is weak, since (with Python 3.14 or `from __future__ annotations`, or via stringifying), just using the class name has the same effect? So `Self` isn't providing any semantic benefits here, it's just a convenience to avoid repeating the class name.

That said, I don't see any reason _not_ to support this; I just think we have more important things to work on. Accepting this but triaging it outside of stable milestone.
